Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -180,19 +180,12 @@ public static bool OverridesEquals(this Type type)
     /// <returns>
     /// Returns <see langword="null"/> if no such property exists.
     /// </returns>
-    public static PropertyInfo FindProperty(this Type type, string propertyName)
+    public static PropertyInfo FindProperty(this Type type, string propertyName, MemberVisibility visibility)
     {
-        while (type != typeof(object))
-        {
-            if (type.GetProperty(propertyName, AllInstanceMembersFlag | BindingFlags.DeclaredOnly) is { } property)
-            {
-                return property;
-            }
-
-            type = type.BaseType;
-        }
+        var properties = type.GetNonPrivateProperties(visibility);
 
-        return null;
+        return properties.FirstOrDefault(p =>
+            p.Name == propertyName || p.Name.EndsWith("." + propertyName, StringComparison.OrdinalIgnoreCase));
```

Comment:
```suggestion
        return Array.Find(properties, p =>
            p.Name == propertyName || p.Name.EndsWith("." + propertyName, StringComparison.OrdinalIgnoreCase));
```

Reply:
I would even suggest to define aan extension Method for this, so that you ca do: properties.Find()

Reply:
Nah, I would only do that when there's multiple uses and it would significantly improve readability.

Reply:
@dennisdoomen In some of my repos, I had multiple places where I did this. If that is not the case here, I fully agree.